src/components/admin/SemesterDetailCard.tsx

```diff
@@ -0,0 +1,79 @@
+"use client";
+
+import { createContext } from "react";
+import Link from "next/link";
+import { Ellipsis } from "lucide-react";
+
+import { Card } from "@/components/Card";
+import { Button } from "@/components/ui/button";
+import {
+  Popover,
+  PopoverContainer,
+  PopoverOpen,
+  PopoverOpenButton,
+} from "../popover";
+import DeleteButton from "./DeleteButton";
+import { EditButton } from "./EditButton";
+
+type DetailCardProps = {
+  id: number;
+  name: string;
+  startDate: string;
+  endDate: string;
+  breakStart: string;
+  breakEnd: string;
+};
+export const ScheduleContext = createContext<DetailCardProps>({});
+
+export const SemesterDetailCard = ({
+  id,
+  name,
+  startDate,
+  endDate,
+  breakStart,
+  breakEnd,
+}: DetailCardProps) => {
+  const SemesterPopover = () => {
+    return (
+      <PopoverContainer>
+        <EditButton />
+        <DeleteButton />
+      </PopoverContainer>
+    );
+  };
```

Comment:
Don't define a component inside another component. This compoenent is re-created everytime the out component is re-rendered which is very bad for performance reasons